util/new_project.sh

```diff
@@ -24,22 +24,43 @@ if [ -e "keyboards/$1" ]; then
 	exit 1
 fi
 
-cd "$(dirname "$0")/.."
+cd "$(dirname "$0")/.." || exit
 
-KEYBOARD_UPPERCASE=$(echo $1 | awk '{print toupper($0)}')
-KEYBOARD_NAME=$(basename $1)
-KEYBOARD_NAME_UPPERCASE=$(echo $KEYBOARD_NAME | awk '{print toupper($0)}')
+KEYBOARD_NAME=$(basename "$1")
+KEYBOARD_NAME_UPPERCASE=$(echo "$KEYBOARD_NAME" | awk '{print toupper($0)}')
+NEW_KBD=keyboards/${KEYBOARD}
 
 
-cp -r quantum/template/base keyboards/$KEYBOARD
-cp -r quantum/template/$KEYBOARD_TYPE/. keyboards/$KEYBOARD
+cp -r quantum/template/base "$NEW_KBD"
+cp -r "quantum/template/$KEYBOARD_TYPE/." "$NEW_KBD"
 
-mv keyboards/${KEYBOARD}/template.c keyboards/${KEYBOARD}/${KEYBOARD_NAME}.c
-mv keyboards/${KEYBOARD}/template.h keyboards/${KEYBOARD}/${KEYBOARD_NAME}.h
-find keyboards/${KEYBOARD} -type f -exec sed -i '' -e "s;%KEYBOARD%;${KEYBOARD_NAME};g" {} \;
-find keyboards/${KEYBOARD} -type f -exec sed -i '' -e "s;%KEYBOARD_UPPERCASE%;${KEYBOARD_NAME_UPPERCASE};g" {} \;
+mv "${NEW_KBD}/template.c" "${NEW_KBD}/${KEYBOARD_NAME}.c"
+mv "${NEW_KBD}/template.h" "${NEW_KBD}/${KEYBOARD_NAME}.h"
+find "${NEW_KBD}" -type f -exec sed -i '' -e "s;%KEYBOARD%;${KEYBOARD_NAME};g" {} \;
+find "${NEW_KBD}" -type f -exec sed -i '' -e "s;%KEYBOARD_UPPERCASE%;${KEYBOARD_NAME_UPPERCASE};g" {} \;
 
-echo "######################################################"
-echo "# /keyboards/$KEYBOARD project created. To start"
-echo "# working on things, cd into keyboards/$KEYBOARD"
-echo "######################################################"
+GIT=$(whereis git)
+if [ "$GIT" != "" ]; then
+  IS_GIT_REPO=$($GIT log >>/dev/null 2>&1; echo $?)
+  if [ "$IS_GIT_REPO" -eq 0 ]; then
```

Comment:
While updating #4222, it occurred to me that this can be written without `test`, since you just want to check the exit code (which is what `if` does by default):
```suggestion
  if $GIT log >/dev/null 2>&1; then  # Check that we're inside a repo
```
This is simpler than the existing code and doesn't have the potentially misleading<sup>†</sup> `"$IS_GIT_REPO" -eq 0` check, but it's not necessarily better since it requires a comment to be as clear as the original.

<sup>†</sup>To clarify, my problem with `"$IS_GIT_REPO" -eq 0` is that to someone who's not used to reading shell scripts, it might look like it's checking that this is _not_ a Git repo, due to the variable name. I think something like `GIT_REPO_STATUS` would be more appropriate.